admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/screen/dukan_managements/DukanManagementViewmodel.kt

```diff
@@ -0,0 +1,159 @@
+package net.thechance.mena.admin_panel.presentation.screen.dukan_managements
+
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import net.thechance.mena.admin_panel.domain.entity.dukan.Dukan
+import net.thechance.mena.admin_panel.domain.exceptions.NoInternetException
+import net.thechance.mena.admin_panel.domain.model.DukanQueryParams
+import net.thechance.mena.admin_panel.domain.model.DukansSortType
+import net.thechance.mena.admin_panel.domain.model.PagedResult
+import net.thechance.mena.admin_panel.domain.model.SortDirection
+import net.thechance.mena.admin_panel.domain.repository.dukan.DukanRepository
+import net.thechance.mena.admin_panel.presentation.base.BaseViewModel
+import net.thechance.mena.admin_panel.presentation.base.ErrorState
+import org.koin.android.annotation.KoinViewModel
+import org.koin.core.annotation.Provided
+import kotlin.uuid.ExperimentalUuidApi
+import kotlin.uuid.Uuid
+
+@OptIn(ExperimentalUuidApi::class)
+@KoinViewModel
+class DukanManagementViewmodel(
+    @Provided
+    private val dukanRepository: DukanRepository,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
+) : BaseViewModel<DukanManagementScreenState, DukanManagementEffect>(
+    DukanManagementScreenState()
+), DukanManagementInteractionListener {
+    private var dukans: List<Dukan> = emptyList()
+    private var searchJob: Job? = null
+
+    init {
+        getDukans()
+    }
+
+    override fun onSearchQueryChange(query: String) {
+        if (query == currentState.query) return
+
+        updateState { it.copy(query = query) }
+        searchJob?.cancel()
+
+        searchJob = viewModelScope.launch {
+            delay(SEARCH_DEBOUNCE_DELAY)
+            updateState { it.copy(pageInfo = it.pageInfo.copy(page = 0)) }
+            getDukans()
+        }
+    }
+
+    override fun onClearQueryClicked() {
+        updateState {
+            it.copy(query = "")
+        }
+        getDukans()
+    }
+
+    override fun onRetryClicked() {
+        getDukans()
+    }
+
+    override fun onPageChanged(page: Int) {
+        updateState { it.copy(pageInfo = it.pageInfo.copy(page = page)) }
+        getDukans()
+    }
+
+    override fun onViewDetailsClicked(dukanId: Uuid) {
+        val dukan = dukans.find { it.id == dukanId } ?: return
+        dukanRepository.storeDukanDetails(dukan)
+        sendEffect(DukanManagementEffect.NavigateToDukanDetails(dukanId))
+    }
```

Comment:
I think you can pass the dukan directly
```suggestion
    override fun onViewDetailsClicked(dukan : Dukan) {
        dukanRepository.storeDukanDetails(dukan)
        sendEffect(DukanManagementEffect.NavigateToDukanDetails(dukan.Id))
    }
```

Reply:
the ui dukan is different from the domain Dukan
@Malak187